runtime/autoload/provider/clipboard.vim

```diff
@@ -55,13 +55,21 @@ endif
 let s:clipboard = {}
 
 function! s:clipboard.get(reg)
-  if s:selections[a:reg].owner > 0
-    return s:selections[a:reg].data
+  let reg = a:reg == '"' ? '+' : a:reg
+  if s:selections[reg].owner > 0
+    return s:selections[reg].data
   end
-  return s:try_cmd(s:paste[a:reg])
+  return s:try_cmd(s:paste[reg])
 endfunction
 
 function! s:clipboard.set(lines, regtype, reg)
+  if a:reg == '"'
+    call s:clipboard.set(a:lines,a:regtype,'+')
+    if s:copy['*'] != s:copy['+']
+      call s:clipboard.set(a:lines,a:regtype,'*')
+    end
+    return 0
```

Comment:
I could make it `return 1` in the end and don't call it for `*` if `set(...,'+')` errored, just to avoid `"clipboard: error when invoking provider"` being printed twice when `unnamedplus,unnamed`, but this should get printed very seldom anyway (as it can't check the exit status,) essentially only if the user `apt-get uninstall xsel` while nvim is running...